cli/cmd/routes.go

```diff
@@ -100,39 +97,46 @@ func renderRouteStats(resp *pb.TopRoutesResponse, options *routesOptions) string
 }
 
 func writeRouteStatsToBuffer(resp *pb.TopRoutesResponse, w *tabwriter.Writer, options *routesOptions) {
-	table := make([]*rowStats, 0)
 
-	for _, r := range resp.GetRoutes().Rows {
-		if r.Stats != nil {
-			route := r.GetRoute()
-			if route == "" {
-				route = defaultRoute
+	tables := make(map[string][]*rowStats)
+
+	for _, resourceTable := range resp.GetOk().GetRoutes() {
+
+		table := make([]*rowStats, 0)
+
+		for _, r := range resourceTable.GetRows() {
+			if r.Stats != nil {
+				route := r.GetRoute()
+				table = append(table, &rowStats{
+					route:       route,
+					dst:         r.GetAuthority(),
+					requestRate: getRequestRate(r.Stats, r.TimeWindow),
+					successRate: getSuccessRate(r.Stats),
+					latencyP50:  r.Stats.LatencyMsP50,
+					latencyP95:  r.Stats.LatencyMsP95,
+					latencyP99:  r.Stats.LatencyMsP99,
+				})
 			}
-			table = append(table, &rowStats{
-				route:       route,
-				dst:         r.GetAuthority(),
-				requestRate: getRequestRate(r.Stats, r.TimeWindow),
-				successRate: getSuccessRate(r.Stats),
-				latencyP50:  r.Stats.LatencyMsP50,
-				latencyP95:  r.Stats.LatencyMsP95,
-				latencyP99:  r.Stats.LatencyMsP99,
-			})
 		}
-	}
 
-	sort.Slice(table, func(i, j int) bool {
-		return table[i].route+table[i].dst < table[j].route+table[j].dst
-	})
+		sort.Slice(table, func(i, j int) bool {
+			return table[i].dst+table[i].route < table[j].dst+table[j].route
+		})
+
+		tables[resourceTable.GetResource()] = table
+	}
 
 	switch options.outputFormat {
 	case "table", "":
-		if len(table) == 0 {
-			fmt.Fprintln(os.Stderr, "No traffic found.  Does the service have a service profile?  You can create one with the `linkerd profile` command.")
-			os.Exit(0)
+		for resource, table := range tables {
+			if len(tables) > 1 {
+				fmt.Fprintf(w, "==> %s <==\t\f", resource)
```

Comment:
Tabwriter requires that all cells be terminated with a \t character.  The \f character is to terminate the column so that this header doesn't influence the ROUTE column beneath it